eventio/base.py

```diff
@@ -88,33 +96,31 @@ def check_size_or_raise(data, expected_length, zero_ok=True):
         raise EOFError('File seems to be truncated')
 
 
-def read_next_header_toplevel(byte_stream):
-    '''Read the next toplevel header object from the file
-    Assumes position of `byte_stream` is at the beginning of a new header.
+def read_sync_marker(byte_stream):
+    '''
+    Read the sync marker from the filehandle.
+    Assumes position of `byte_stream` is at the beginning of a new toplevel header.
 
     Raises stop iteration if not enough data is available.
+    Raises NotImplementedError if BigEndian sync marker is encountered
     '''
     sync = byte_stream.read(constants.SYNC_MARKER_SIZE)
-    check_size_or_raise(sync, constants.SYNC_MARKER_SIZE)
+    check_size_or_raise(sync, constants.SYNC_MARKER_SIZE, zero_ok=True)
+
     endianness = parse_sync_bytes(sync)
 
-    return read_next_header_sublevel(byte_stream, endianness)
+    if endianness == '>':
+        raise NotImplementedError(
+            'Big endian byte order is not supported by this reader'
+        )
 
 
-def read_next_header_sublevel(byte_stream, endianness, parent_address=0):
-    '''Read the next sublevel header object from the file
+def read_header(byte_stream, offset, toplevel=False):
+    '''Read the next header object from the file
     Assumes position of `byte_stream` is at the beginning of a new header.
 
-    endianness: char
-        '<' or '>' for little or big endiannes respectively.
-        Is either read from next header or already known.
-
```

Comment:
You are totally right, about getting rid of this endianness stuff here. We raise early and are done with it. Very nice. I like it a lot.